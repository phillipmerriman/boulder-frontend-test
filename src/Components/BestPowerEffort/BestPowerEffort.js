import React from 'react';

const BestPowerEffort = (props) => {
  const { twentyMinute, oneMinute } = props;

  return (
    <>
      <h2 data-testid='bestOneMinute'>Best 1 Minute Power Effort: {oneMinute}</h2>
      <h2 data-testid='bestTwentyMinute'>Best 20 Minute Power Effort: {twentyMinute}</h2>
    </>
  );
};

export default BestPowerEffort;
