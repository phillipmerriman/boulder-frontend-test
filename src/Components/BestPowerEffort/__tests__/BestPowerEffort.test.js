import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import BestPowerEffort from '../BestPowerEffort';

describe('BestPowerEffort component', () => {

    afterEach(cleanup);
    
    it('renders without crashing', () => {
        const div = document.createElement("div");
        ReactDOM.render(<BestPowerEffort></BestPowerEffort>, div);
    });
    
    it('renders best 1 minute power effort correctly', () => {
        render(<BestPowerEffort oneMinute={300}></BestPowerEffort>);
        expect(screen.getByTestId('bestOneMinute')).toHaveTextContent(300);
    });
    
    it('renders best 20 minute power effort correctly', () => {
        render(<BestPowerEffort twentyMinute={200}></BestPowerEffort>);
        expect(screen.getByTestId('bestTwentyMinute')).toHaveTextContent(200);
    });
    
    it('matches snapshot', () => {
        const tree = renderer.create(<BestPowerEffort twentyMinute={200} oneMinute={300}></BestPowerEffort>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});
