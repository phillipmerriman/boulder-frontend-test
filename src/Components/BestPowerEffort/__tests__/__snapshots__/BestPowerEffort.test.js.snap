// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestPowerEffort component matches snapshot 1`] = `
Array [
  <h2
    data-testid="bestOneMinute"
  >
    Best 1 Minute Power Effort: 
    300
  </h2>,
  <h2
    data-testid="bestTwentyMinute"
  >
    Best 20 Minute Power Effort: 
    200
  </h2>,
]
`;
